### DISCUSSION TITLE
zxing : set two button to two textview

### DISCUSSION THREAD

**Discussion by jeredez182** [journeyapps/zxing-android-embedded#601] [Q&A]
hi ! i'm on 3.6.0 but your answer dont fit to me, here is my mainactivity, i just dont see how to set up the requestcode in the last part, to call result 1 from button 1 and result 2 from button 2 in allocated textview...thanks for your help

public class MainActivity extends AppCompatActivity {

Button btn_scan;
TextView text_result;
Button btn_scan2;
TextView text_result2;

@Override
protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    text_result = findViewById(R.id.text_result);
    text_result2 = findViewById(R.id.text_result2);
  
    findViewById(R.id.btn_scan).setOnClickListener(new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            IntentIntegrator integrator = new IntentIntegrator(MainActivity.this);
            integrator.setDesiredBarcodeFormats(IntentIntegrator.ONE_D_CODE_TYPES);
            integrator.setPrompt("Scan);
            integrator.setCameraId(0);  // Use a specific camera of the device
            integrator.setBeepEnabled(true);
            integrator.setBarcodeImageEnabled(true);
            integrator.initiateScan();
        }
    });

    findViewById(R.id.btn_scan2).setOnClickListener(new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            IntentIntegrator integrator = new IntentIntegrator(MainActivity.this);
            integrator.setDesiredBarcodeFormats(IntentIntegrator.ONE_D_CODE_TYPES);
            integrator.setPrompt("Scan");
            integrator.setCameraId(0);  // Use a specific camera of the device
            integrator.setBeepEnabled(true);
            integrator.setBarcodeImageEnabled(true);
            integrator.initiateScan();
        }
    });
  
       }

@Override
protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {

    IntentResult result = IntentIntegrator.parseActivityResult(requestCode, resultCode, data);
    if(result != null) {
        if(result.getContents() == null) {
            Toast.makeText(this, "cancel", Toast.LENGTH_LONG).show();
        } else {
            //Toast.makeText(this, "scan: " + result.getContents(), Toast.LENGTH_LONG).show();
            text_result.setText(""+result.getContents());
        }
    } else {
        super.onActivityResult(requestCode, resultCode, data);
    }
}

**Comment by rkistner** [comment]
To use a different request code, see this example:

https://github.com/journeyapps/zxing-android-embedded/blob/13440ad8751dcaba4177fa50dfee639dac52a53d/sample/src/main/java/example/zxing/MainActivity.java#L36

https://github.com/journeyapps/zxing-android-embedded/blob/13440ad8751dcaba4177fa50dfee639dac52a53d/sample/src/main/java/example/zxing/MainActivity.java#L114-L142

It's important to use the `IntentIntegrator.parseActivityResult(resultCode, data)` version, _without_ the `requestCode`, otherwise any non-default requestCode will be filtered out.
